src/entrypoints/entrypoints.ts

```diff
@@ -34,19 +34,24 @@ export class NetworkEntrypoint {
     constructor(options: {
         networkProviderUrl: string;
         networkProviderKind: string;
+        networkProviderOptions?: NetworkProviderConfig;
         chainId: string;
         clientName?: string;
         withGasLimitEstimator?: boolean;
         gasLimitMultiplier?: number;
     }) {
+        options.networkProviderOptions = options.networkProviderOptions || {};
+
+        if (options.clientName) {
+            options.networkProviderOptions.clientName = options.clientName;
+        } else if (options.networkProviderOptions?.headers?.["User-Agent"]) {
+            options.networkProviderOptions.clientName = options.networkProviderOptions.headers["User-Agent"];
```

Comment:
This is an opaque trick (a hack), and assumes that "client name" and "header user agent" are synonyms for network providers (somehow, it breaks a bit their encapsulation). 

Maybe drop this logic, and, if desired, a separate PR can be made, but not for the entrypoints. See the file below for a possible place of adjustment:
 - https://github.com/multiversx/mx-sdk-js-core/blob/main/src/networkProviders/userAgent.ts